tests/e2e/build_test.go

```diff
@@ -0,0 +1,91 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"golang.org/x/net/context"
+)
+
+// Kedge location
+var KedgeLoc = "$GOPATH/src/github.com/kedgeproject/kedge"
+
+func runCmd(cmdS string) ([]byte, error) {
+	var cmd *exec.Cmd
+	var out, stdErr bytes.Buffer
+	cmd = exec.Command("/bin/sh", "-c", cmdS)
+
+	cmd.Stdout = &out
+	cmd.Stderr = &stdErr
+
+	err := cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("error running command %v: %s", cmd, err)
+
+	}
+	return out.Bytes(), nil
+}
+
+func TestBuild(t *testing.T) {
+	// Context Dir to be passed for kedge build
+	var contextDir = KedgeLoc + "/docs/examples/build"
+	var (
+		err error
+	)
+
+	cli, err := client.NewEnvClient()
+	if err != nil {
+		t.Error(err)
+	}
+
+	dockerRegStart := "docker run -d -p 5000:5000 --restart=always --name kedge_registry registry:2"
+	reg, err := runCmd(dockerRegStart)
+	if err != nil {
+		panic(err)
+	}
+
+	t.Logf("Local docker registry has been set up: %s", reg)
+
+	runKedgeCmd := KedgeLoc + "/kedge build -c " + contextDir + " -i localhost:5000/test:2.0 -p"
+
+	t.Logf("Running '%s'\n", runKedgeCmd)
+	kedgeBuild, err := runCmd(runKedgeCmd)
+	if err != nil {
+		t.Fatalf("Error running kedge: %v", err)
+	}
+
+	t.Log(string(kedgeBuild))
+
+	images, err := cli.ImageList(context.Background(), types.ImageListOptions{})
+	if err != nil {
+		t.Error(err)
+	}
+
+	for _, image := range images {
+		if image.RepoTags[0] == "localhost:5000/test:2.0" {
+			t.Log("The image has been uploaded successfully.")
+
+			rmc := "docker rmi " + image.ID
+
+			_, err := runCmd(rmc)
+			if err != nil {
+				t.Error(err)
+			}
+		}
+	}
+
+	// Delete the registry
+	_, err = runCmd("docker stop kedge_registry")
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = runCmd("docker rm -v kedge_registry")
+	if err != nil {
+		t.Error(err)
+	}
```

Comment:
put all of the clean up code in a fucntion and then call it with defer so that we always do the clean up!